postgres/user_service_test.go

```diff
@@ -24,65 +24,65 @@ func TestUserService(t *testing.T) {
 	ctx := context.Background()
 
 	t.Run("ok: create user", func(t *testing.T) {
-		email1, err := core.NewEmailAddress(tests.Faker.Email())
+		email1, err := core.ParseEmailAddress(tests.Faker.Email())
 		tests.Check(err)
-		var email2 core.EmailAddress
+		var email2 *core.EmailAddress
 		// Make sure both e-mail addresses are unique
 		for email2 == nil || email1.String() == email2.String() {
-			email2, err = core.NewEmailAddress(tests.Faker.Email())
+			email2, err = core.ParseEmailAddress(tests.Faker.Email())
 			tests.Check(err)
 		}
 		for _, data := range []CreateUserTest{
 			{
 				name:        tests.Faker.Name(),
-				email:       email1,
+				email:       *email1,
```

Comment:
Het is perfect ok om dit zo aan te passen, maar het had een stuk simpeler geweest om het resultaat van de parse een andere naam te geven (bv "temp") en dan `email1 := *temp` ofzo te doen, ik denk dat een mogelijke error over een nil pointer dereference zo ook duidelijker zou geweest zijn dan die nu zou zijn moest dat voorkomen (een expliciete nil check is uiteraard altijd nog duidelijker) 😛

Reply:
Het probleem is dat je niet zeker weet in welke volgorde die testen uitgevoerd worden (zeker aangezien deze eigenlijk allemaal perfect parallel zouden kunnen zijn), en de errors als je daar nil krijgt zijn heel moeilijk leesbaar. Je kan er binnen een test ook niet echt van uit gaan dat een andere test die case al nakijkt omdat die andere test mogelijks verwijderd of gerefactord kan worden en `nil, nil` dan helemaal niet meer getest zou worden (tenzij daar echt een aparte test voor bestaat ofzo). Je hoeft dat hier niet aan te passen, ik wil gewoon mijn filosofie daarrond duidelijk maken 😛  Dit is de kern van wat ik bedoel: het is in ieder denkbaar geval robuuster en duidelijker als je alle pointers die je wilt dereferencen eerst assert met NotNil, onafhankelijk van wat andere tests al dan niet doen of wat je weet over de interne werking van bepaalde functies. Ik kan mij momenteel geen situatie inbeelden waarin dat niet zo zou zijn 😛

Reply:
Goed punt, ik pas het niet aan, maar probeer er aan te denken voor de toekomst!